session.cpp

```diff
@@ -39,11 +39,10 @@ int session::handle_request(const boost::system::error_code& error,
     return -1;
   }
 
-  // TODO: pls put out this fire
-  // // find longest prefix that uniquely matches uri (our key for handler_map)
-  // std::string longest_prefix = get_longest_prefix(request->uri());
-  // // get corresponding handler
-  // std::shared_ptr<RequestHandler> handler_ptr = handler_map[longest_prefix];
+  // find longest prefix that uniquely matches uri (our key for handler_map)
+  std::string longest_prefix = get_longest_prefix(request->uri());
+  // get corresponding handler
+  std::shared_ptr<RequestHandler> handler_ptr = handler_map[longest_prefix];
```

Comment:
we should add a handler in this map of the following:

"" -> default handler

Since get_longest_prefix returns empty string if it finds no match